src/inet/AsyncDNSResolverSockets.cpp

```diff
@@ -87,14 +110,14 @@ INET_ERROR AsyncDNSResolverSockets::Shutdown(void)
     INET_ERROR err = INET_NO_ERROR;
     int pthreadErr;
 
-    AsyncMutexLock();
-
-    mInet->State = InetLayer::kState_ShutdownInProgress;
+    {
+        LockHolder asyncMutexLock(*this);
```

Comment:
This illustrates one of the downsides of this pattern--the point at which the lock is acquired doesn't read as an imperative statement, and hence is easily missed by the unaware reader.

I think a construct like this would be more obvious:

```
{
    auto lock = LockHolder::AquireLock(*this);

    ...
}
```